### ISSUE TITLE
Different random number seeds affect the results

### ISSUE THREAD

**Issue by kdslsdf** [AdityaLab/pinnsformer#6]
Hello, I have a strong interest and appreciation for your research, so I have tried my best  to reproduce the numerical experiments presented in the paper using the code provided in this repository.

But when reproducing, I found that pinnsformer has significant differences in rMAE and rRMSE under different random number seeds. For example, for the convergence equation, when seed=66, rRMSE can reach 0.13, but in other seed cases it is only about 0.6. My reproduction code is here.(https://gist.github.com/kdslsdf/469c0d5f99ebf6056fb3dc9087205f11)

May I ask if you can explain this situation? I would greatly appreciate your response.

**Comment by tsano430** [comment]
I also encountered the same issue and raised an issue earlier (https://github.com/AdityaLab/pinnsformer/issues/5).

I don't think this is effective for all cases, but in the governing equation I was working on, I avoided the situation by switching from Adam to L-BFGS. Thank you.

**Comment by kdslsdf** [comment]
> I also encountered the same issue and raised an issue earlier (#5).
> 
> I don't think this is effective for all cases, but in the governing equation I was working on, I avoided the situation by switching from Adam to L-BFGS. Thank you.

Can training with Adam first and then L-BFGS avoid the influence of random number seeds? I will try to conduct experiments to verify. Anyway, thank you for your response.

**Comment by leozhao1997** [comment]
Thanks for the discussion. 

Here is the dependency that may help reproduce the results (can be slightly different from the notebook results as I kind of forgot to export the environment at that time, and there could be some auto-updates, and also cuda version could matter)

dependencies:
  - python=3.10.4
  - pytorch=2.0.1
  - pytorch-cuda=11.7
  - numpy=1.22.4
  - cudatoolkit=11.6.0
  
Also, you might try some hyperparameter tuning. My experience is that it is more sensitive on 1d convection than 1d reaction and others.
